### DISCUSSION TITLE
Sparse matrix in cvxpy

### DISCUSSION THREAD

**Discussion by UzTak** [cvxpy/cvxpy#2314] [Q&A] [answered]
I am trying to solve semidefinite programming that involves a large block diagonal matrix like below. 
<img width="421" alt="image" src="https://github.com/cvxpy/cvxpy/assets/49874360/6cf68980-a220-4a49-84dd-061851558ed1">
Each K_i is a matrix with a size of nu * nx.
Currently, I am setting up this block diagonal matrix as follows. 

```python
import numpy as np 
import cvxpy as cvx

k = [cvx.Variable((nu, nx)) for j in range(PS.N)]
K = k[0]
for i in range(N-1):
    K = cvx.bmat([[K, np.zeros((nu*(i+1), nx))], 
                     [np.zeros((nu, nx*(i+1))), k[i+1]]])
K = cvx.bmat([[K, np.zeros((nu*N, nx))]]) 
```

However, this results in a huge dimension in the design variable when scalarized, making the computation almost intractable. (cf. currently I am setting N = 80, nx = 4, and nu = 2).
Is there any good way to define the matrix variable K using sparsity matrix from scipy package? 

I am also trying to write something like this, while this returns a ValueError :
` "ValueError: object dtype is not supported by sparse matrices" due to the type of data1. `

```python 
from scipy.sparse import csr_array

row = np.repeat(np.arange(N*nu), nx)
col = np.concatenate([np.tile(np.arange(start * nx, (start + 1) * nx), nu) for start in range(N)])
data1 = [cvx.Variable(1) for _ in range(N*nu*nx)]
array = csr_array((data1, (row, col)), shape=(N*nu, N*nx))
```

I appreciate your input, thank you!

**Comment by Transurgeon** [comment, marked as answer]
Unfortunately it is not yet possible to define a sparse variable matrix using CVXPY. 
See the similar discussion question [here](https://github.com/cvxpy/cvxpy/discussions/2039). 
I suspect that you could check the leading minors of the diagonal and make sure those are PSD, which in turn might satisfy that the whole block diag matrix is also PSD, however, I am not too certain how accurate that is. See a related question [here](https://math.stackexchange.com/questions/904454/is-this-symmetric-block-diagonal-matrix-positive-semi-definite).
